src/Planners/src/UnicycleTrajectoryPlanner.cpp

```diff
@@ -0,0 +1,841 @@
+        {
+            outputVect.push_back(iDynTree::toEigen(v));
+        };
+
+        return outputVect;
+    };
+
+    std::vector<Eigen::Vector2d> dcmPosition, dcmVelocity;
+    dcmPosition = convertToEigen(dcmGenerator->getDCMPosition());
+    dcmVelocity = convertToEigen(dcmGenerator->getDCMVelocity());
+    m_pImpl->output.dcmTrajectory.dcmPosition = dcmPosition;
+    m_pImpl->output.dcmTrajectory.dcmVelocity = dcmVelocity;
+
+    // get the CoM planar trajectory
+    auto time = std::chrono::nanoseconds(static_cast<int>(initTime * 1e9));
+    Eigen::Vector4d state;
+    state.head(2) = m_pImpl->input.comInitialState.initialPlanarPosition;
+    state.tail(2) = m_pImpl->input.comInitialState.initialPlanarVelocity;
+    m_pImpl->comSystem.dynamics->setState({state.head(4)});
+    using namespace BipedalLocomotion::GenericContainer::literals;
+    auto stateDerivative = BipedalLocomotion::GenericContainer::make_named_tuple(
+        BipedalLocomotion::GenericContainer::named_param<"dx"_h, Eigen::VectorXd>());
+    Eigen::Vector4d controlInput;
+    std::vector<Eigen::Vector2d> comPlanarPosition, comPlanarVelocity, comPlanarAcceleration;
+
+    for (size_t i = 0; i < dcmPosition.size(); i++)
+    {
+        // populate CoM planar position
+        comPlanarPosition.push_back(state.head<2>());
+
+        // set the control input, u
+        controlInput << dcmPosition.at(i), dcmVelocity.at(i);
+        m_pImpl->comSystem.dynamics->setControlInput({controlInput});
+
+        // compute the state derivative xdot = Ax + Bu
+        m_pImpl->comSystem.dynamics->dynamics(time, stateDerivative);
+
+        // populate CoM planar velocity and acceleration
+        comPlanarAcceleration.push_back(stateDerivative.get_from_hash<"dx"_h>().tail<2>());
+        comPlanarVelocity.push_back(stateDerivative.get_from_hash<"dx"_h>().head<2>());
+
+        // advance the integrator for one step
+        m_pImpl->comSystem.integrator->oneStepIntegration(time,
+                                                          std::chrono::nanoseconds(
+                                                              static_cast<int>(dt * 1e9)));
+        state.head(4) = std::get<0>(m_pImpl->comSystem.integrator->getSolution());
+
+        // update the system state
+        m_pImpl->comSystem.dynamics->setState({state});
+        time += std::chrono::nanoseconds(static_cast<int>(dt * 1e9));
+    }
+
+    // get the CoM height trajectory
+    std::vector<double> comHeightPosition, comHeightVelocity, comHeightAcceleration;
+    auto comHeightGenerator = m_pImpl->generator.addCoMHeightTrajectoryGenerator();
+    comHeightGenerator->getCoMHeightTrajectory(comHeightPosition);
+    comHeightGenerator->getCoMHeightVelocity(comHeightVelocity);
+    comHeightGenerator->getCoMHeightAccelerationProfile(comHeightAcceleration);
+
+    // stack the CoM planar and the height trajectory
+    std::vector<Eigen::Vector3d> comPosition, comVelocity, comAcceleration;
```

Comment:
Same as before

Reply:
My only concern is that I did not want to duplicate information that are stored also in the output as for the example `m_pImpl->output.comTrajectory.position`. For code readability instead of using directly `m_pImpl->output.comTrajectory.position`, I preferred to use `comPosition` for intermediate operations and then assign it with `m_pImpl->output.comTrajectory.position = comPosition`.

Also, it might happen that the `avance` member function is not called at every time instant. For example this is what happens with `UnicycleTrajectoryGenerator` of #845, which calls the `advance` function of `UnicycleTrajectoryPlanner` only at certain time instants (i.e. when a new trajectory is required).

But, I am open to modify it if you think it makes it clearer and/or faster.

Reply:
same as https://github.com/ami-iit/bipedal-locomotion-framework/pull/844#discussion_r1604525894